go/vt/vttablet/tabletserver/health_streamer.go

```diff
@@ -349,7 +349,12 @@ func (hs *healthStreamer) reload() error {
 
 	views, err := hs.getChangedViewNames(ctx, conn)
 	if err != nil {
-		return err
+		if len(tables) == 0 {
+			return err
+		}
+		// there are some tables, we need to send that in the health stream.
```

Comment:
what if the inverse happens - there are view changes, and an error is found when looking for changed tables? don't we want to send the view changes through?

Reply:
The reason why we have to send the tables changes is because if we skip this time then next time those changed tables would not be reflected. 
Views change checking happens latter so it is not the case for that. 
The only reason tables will fail if there is some problem with deployment or the connection is temporarily unavailable or vttablet shutting down in all such cases it is ok to wait for the next iteration.